include/swift/IRGen/Linking.h

```diff
@@ -110,6 +110,14 @@ class LinkEntity {
     /// A swift metaclass-stub reference.  The pointer is a ClassDecl*.
     SwiftMetaclassStub,
 
+    /// A class metadata base offset global variable.  This stores the offset
+    /// of the immediate members of a class (generic parameters, field offsets,
+    /// vtable offsets) in the class's metadata.  The immediate members begin
+    /// immediately after the superclass members end.
```

Comment:
If this is the only global offset, does this mean we need to know the complete layout of the immediate members in order to use any of it?  I guess we won't be using this outside the module for methods, and we'd only use it for properties if the class was fixed-layout for some reason (but with a depenently-offsetted property), so maybe it only affects generic requirements, and we can assume this points directly at those.

Reply:
Yeah, when the complete resilience model is implemented this will only be used for generic arguments. Right now it's also going to be used for virtual methods in super calls and overrides, but I'll implement the rest soon.